sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SyntheticDataPubSubPublisher.java

```diff
@@ -83,14 +97,41 @@ public static void main(String[] args) throws IOException {
 
     Pipeline pipeline = Pipeline.create(options);
 
-    pipeline
-        .apply("Read synthetic data", Read.from(new SyntheticBoundedSource(sourceOptions)))
-        .apply("Map to PubSub messages", MapElements.via(new MapBytesToPubSubMessage()))
-        .apply("Write to PubSub", PubsubIO.writeMessages().to(options.getInsertionPipelineTopic()));
-
+    if (!options.getKafkaBootstrapServerAddress().isEmpty()) {
+      pipeline
+          .apply("Read synthetic data", Read.from(new SyntheticBoundedSource(sourceOptions)))
+          .apply("Map to Kafka messages", MapElements.via(new MapKVToString()))
+          .apply(
+              "Write to Kafka",
+              KafkaIO.<Void, String>write()
+                  .withBootstrapServers(options.getKafkaBootstrapServerAddress())
+                  .withTopic(options.getInsertionPipelineTopic())
+                  .withValueSerializer(StringSerializer.class)
+                  .values());
+    } else {
+      pipeline
+          .apply("Read synthetic data", Read.from(new SyntheticBoundedSource(sourceOptions)))
+          .apply("Map to PubSub messages", MapElements.via(new MapBytesToPubSubMessage()))
+          .apply(
+              "Write to PubSub", PubsubIO.writeMessages().to(options.getInsertionPipelineTopic()));
+    }
     pipeline.run().waitUntilFinish();
   }
 
+  private static class MapKVToString extends SimpleFunction<KV<byte[], byte[]>, String> {
+    @Override
+    public String apply(KV<byte[], byte[]> input) {
+      StringBuilder stringBuilder =
+          new StringBuilder()
```

Comment:
Hint: an "oneliner" for this can be done using `String.format("{%s,%s}", Arrays.toString(input.getKey()), Arrays.toString(input.getValue()))`